📖 Update release process for 'mystmd' to include PyPI and Conda (#2667)
executablebooks/mystmd: docs/developer.md

`mystmd` is published to NPM via a GitHub Action. Here's how to publish a release.

- **Find the changesets PR**. This contains a list of the version updates that will be included with this release. [Here's an example of a release PR](https://github.com/jupyter-book/mystmd/pull/1896).
- Review the changeset PR.
  - Ensure that `mystmd` is in the changesets. Or you are intentially **not** releasing `mystmd` (this generally shouldn't happen), in which case the python and release notes are expected to fail.
  - Ensure that private or non-existent packages like docs, etc. are not in the changesets (these will cause an early failure)
  - Ensure that there are no **new** myst packages that need to be created/linked
- **Merge the changesets PR**. After merging that PR, [this GitHub action will make a release](https://github.com/jupyter-book/mystmd/blob/main/.github/workflows/release.yml).
  - It calls `npm run version` to generate the changelog (to review the changelog, you can run that command locally too).
  - It then publishes the updated packages to the [`mystmd` npm registry](https://www.npmjs.com/package/mystmd) (it calls `npm run publish:ci`, which calls `changeset publish`).

>   - It creates multiple git version tag (which you'll use in making the GitHub release).

Author: ```suggestion
  - It creates multiple git version tags (which you'll use in making the GitHub release).
```

  - It releases to PyPI
  - The release notes are created automatically based on the `mystmd@v...` tag
- Confirm that NPM, PyPI versions are released and a GitHub release has been made. Once this happens and there are no errors, you're done!
- A short time later, a Conda PR is opened automatically and needs to be merged.

> If there are errors, ask for help in `release_coordination` on discord. The exact fix will depend where in the automated release things broke. Some tips:

Reviewer: ```suggestion
##### Fixing errors during a release

Ask for help in `release_coordination` on discord. The exact fix will depend where in the automated release things broke. Some tips:
```

> - In general, do not revert a release PR, instead fix the errors (permissions, etc. with NPM/tokens).

Author: Yep!
